Research inventory

lomamech opened the pull request:
Filter Inventory list by Inventory Group
- 	Update inventory service adding formatFilter Parameters
-      	Resolve filter between Inventory configuration page and inventory registry
-      	Adding test integration for inventory research in the server
-      	Adding test E2E to filter Inventory registry with button Search 

#1147

lomamech committed:
Filter Inventory list by Inventory Group
Update Inventory registry adding button for the reaserch
Update InventoryService adding of Modal for reaserch
Adding request for filtering Inventory

lomamech committed:
Update inventory service adding formatFilter Parameters
Adding headers of filtered element on inventory registry

lomamech committed:
Resolve filter between Inventory configuration page and inventory registry

lomamech committed:
Adding test integration for inventory research in the server

lomamech committed:
Adding test E2E to filter Inventory registry with button Search
Update button research adding an ID

lomamech committed:
Use fa-close for clear filters

jniles commented on client/src/partials/inventory/list/list.html:
```diff
@@ -27,15 +27,45 @@
+    class="text-danger"
+    data-method="clear">
+    <i class="fa fa-close text-danger"></i>
+    {{ "FORM.INFO.CLEAR_FILTERS" | translate }}
```
This code must use the `translate` directive.

jniles commented on client/src/partials/inventory/list/modals/search.modal.html:
```diff
@@ -0,0 +1,70 @@
+<form
+  name="ModalForm"
+  ng-submit="ModalCtrl.submit(ModalForm)"
+  data-modal="patient-search"
```
You can delete this attribute - it is not used.

jniles commented on client/src/partials/inventory/list/modals/search.modal.html:
```diff
@@ -0,0 +1,70 @@
+  bh-form-defaults
+  novalidate>
+
+  <div class="modal-header" style="box-shadow : 0px 2px 4px -4px black; z-index:1000">
```
Why does this have such complicated styling?

jniles commented on client/src/partials/inventory/list/modals/search.modal.html:
```diff
@@ -0,0 +1,70 @@
+    </ol>
+  </div>
+
+  <div class="modal-body" style="max-height : 400px; overflow: auto;">
```
Just a note - we should standardize these max-heights in `bhConstants`.  I'll make an issue about it.

jniles commented on client/src/partials/inventory/list/modals/search.modal.js:
```diff
@@ -0,0 +1,59 @@
+ *
+ * @description
+ * This controller is responsible for setting up the filters for the patient
+ * search functionality on the patient registry page.  Filters that are already
```
This description should be updated to fit the Inventory Search code.

jniles commented on client/src/partials/inventory/list/modals/search.modal.js:
```diff
@@ -0,0 +1,59 @@
+  vm.cancel = cancel;
+  vm.clear = clear;
+
+  InventoryGroups.Groups.read()
```
The service used here is actually the `InventoryService`.  Renaming it `InventoryGroups` is misleading.  Please change the name so that this code reads:

```js
Inventory.Groups.read()
```

jniles commented on server/controllers/inventory/inventory/core.js:
```diff
@@ -153,6 +155,33 @@ function getItemsMetadataById(uuid) {
+    db.bid(params.group_uuid);  
+  }
+  
+  let filters = new FilterParser(params, { tableAlias : 'inventory' });
```
Good use of filter parser!

jniles commented on server/controllers/inventory/inventory/core.js:
```diff
@@ -153,6 +155,33 @@ function getItemsMetadataById(uuid) {
 
+
+function getItemsMetadataSearch(params){
+  if(params.group_uuid){
```
I think filter parser will automatically do this conversion for you.  Could you try?

jniles commented on server/controllers/inventory/inventory/core.js:
```diff
@@ -153,6 +155,33 @@ function getItemsMetadataById(uuid) {
+  
+  let filters = new FilterParser(params, { tableAlias : 'inventory' });
+
+  var sql =
```
Please use `const` or `let` instead of `var`.

jniles reviewed (commented):
@lomamech, this is a really good PR.  I've made a few comments to bring the code in line with the latest standard and remove confusing/extra lines.

After that, LGTM.

lomamech committed:
Fix code
Using good directive for translation
Remove not necessary style
Update a text of comment
Using of let or const instead var in inventory/core.js
